compiler/semtypes.nim

```diff
@@ -227,6 +227,7 @@ proc isRecursiveType(t: PType, cycleDetector: var IntSet): bool =
 
 proc fitDefaultNode(c: PContext, n: PNode): PType =
   let expectedType = if n[^2].kind != nkEmpty: semTypeNode(c, n[^2], nil) else: nil
+  n[^1] = semExprWithType(c, n[^1], {efTypeAllowed}, expectedType)
```

Comment:
Why the `efTypeAllowed` here? And isn't semchecking already done by `semConstExpr`?

Reply:
Seems that overloading resolving needs to be done by `semExpr`. `efTypeAllowed` is used to make the type mismatch error message better. I suppose I can remove it.

Reply:
You're right, I didn't notice that.